input/input.go

```diff
@@ -89,7 +92,9 @@ func (in DefaultHandler) ProcessMetricData(md *schema.MetricData, partition int3
 		log.Debugf("in: Invalid metric %v: %s", md, err)
 		return
 	}
-	if md.Time == 0 {
+	// in cassandra we store timestamps and interval as 32bit signed integers.
+	// math.MaxInt32 = Jan 19 03:14:07 UTC 2038
+	if md.Time == 0 || md.Time >= math.MaxInt32 || md.Interval >= math.MaxInt32 {
```

Comment:
I'm assuming that Time & Interval will only be `>math.MaxInt32` if the received data is somehow corrupted. If we're now filtering corrupt data here we should probably also check if they are `<0`

Reply:
good point, but that should be added to the schema repository.
the validation here is only for additional validation due to MT's limitations

Reply:
I'd still change from `== 0` to `<= 0` here on the off chance something strange happens (and log the received value).